File: CPP-Version/Sort/insertion_sort.cpp
#include <iostream>
#include <sstream>
#include <vector>

using namespace std;

vector<int> arr_comprehension()
{
    cout << "Input Array elements: ";
    vector<int> arr;
    string input;
    getline(cin, input);

    stringstream ss(input);
    int num;
    while (ss >> num)
    {
        arr.push_back(num);
    }

    return arr;
}

vector<int> insertion_sort(vector<int> arr)
{
    for (int i = 1; i < size(arr); i++)
    {
        int val = arr[i];
        int idx = i - 1;
        while (idx >= 0 && arr[idx] > val)
        {
            arr[idx + 1] = arr[idx];
            idx -= 1;
        }
        arr[idx + 1] = val;
    }
    return arr;
}

int main()
{
    vector<int> arr = arr_comprehension();
    vector<int> req_arr = insertion_sort(arr);
    cout << endl
         << "Sorted array is as follows:" << endl;
    for (int i : req_arr)
    {
        cout << i << "\t";
    }

    return 0;
}

File: Sorting algorithm/insertion_sort.py
def ins_srt(lst):
	for i in range(1, len(lst)):
		val = lst[i]
		idx = i - 1
		while idx >= 0 and lst[idx] > val:
			lst[idx + 1] = lst[idx]
			idx -= 1
		lst[idx + 1] = val
	return lst

arr = [int(i) for i in input().split()]

print("Sorted array is: ", ins_srt(arr))